### PR TITLE
feat: set cursor and extended colours

### LINKED ISSUE #17: Add Cursor Color
Instead of using the default cursor colors (which are just foreground and background inverted) the catppuccin themes for other terminal emulators such as kitty and alacritty use rosewater as the cursor color and keep base for the text

So I'd like to propose the same for the foot themes

This should be as easy as adding
```
[cursor]
color={{base.hex}} {{rosewater.hex}}
```
to the foot.tera file

### LINKED ISSUE #18: Add Extended Colors
Add color 16 (peach) and color 17 (rosewater) in accordance with the style guide.

### PR DESCRIPTION
**Pull request by ghost**
Implements #17 and #18

The extended colors are pretty straightforward.

For the cursor colors however, other terminal emulators like kitty and alacritty use Base as the Cursor Text color for every flavor, while the official [style guides](https://github.com/catppuccin/catppuccin/blob/main/docs/style-guide.md) specify Base for Latte and Crust for all the other flavors.

As it stands right now I used Base for every flavor but I'd like the input of other maintainers on whether we should align ourselves with other terminal emulators or strictly follow the guidelines when they exist.

Edit:
After carefully observing kitty and alacritty's .tera files I've noticed some things that diverge from the style guide, like selection fg/bg being Base/Rosewater rather than Text/Overlay2(30% opacity) and URLs being Rosewater rather than Blue.

Since foot already follows the style guidelines closely I'd be up for editing my PR to match them rather than following other terminal emulators like initially suggested.

### CODE REVIEW

**Comment by ghost** [comment]
pinging @sgoudham for visibility since there is no active maintainer

**Review by sgoudham — APPROVED** [review]
Hey 👋,
Thanks for the PR and apologies for the wait on the review. 

I'm not exactly proud of the terminal colours at the moment and we are actively thinking about how to improve them in https://github.com/catppuccin/catppuccin/issues/1961

It would be good to align to the style guide and I can see that you've edited the PR to match that, thank you. What's interesting is that Latte specifies Rosewater/Base for the cursor colours but I've just seen that Rosewater/Text yields a tiny bit more contrast. 

In any case, the colours do need to be revamped for sure. I'm happy with accepting the PR as it is and thanks again for updating it!

**Pull request merged**
